File: Array/704. Binary Search/answer.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        int val;
        int idx;

        while (high >= low) { // contains == because [high == low] might contain target value
            idx = (high + low) / 2;
            val = nums[idx];
            if (val == target) {
                return idx;
            }
            //not target, move depending on num
            else if (val > target) {
                //move high
                high = idx - 1;
            }
            else {
                low = idx + 1; //
            }
        }
        return -1;
    }
}

File: Array/704. Binary Search/answer.py


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while right >= left:
            index = (right + left) // 2
            if nums[index] == target:
                return index
            elif nums[index] > target:
                right = index - 1
            else:
                left = index + 1
        return -1

